pkg/convert/pprof/streaming/vt_location.go

```diff
@@ -10,12 +11,15 @@ type locationFunctions struct {
 	lines []line
 }
 
-func (l *locationFunctions) reset() {
+func (l *locationFunctions) reset(a arenahelper.ArenaWrapper, nLocations int) {
+	if l.lines == nil {
+		l.lines = arenahelper.MakeSlice[line](a, 0, nLocations*2)
+	}
 	l.lines = l.lines[:0]
```

Comment:
I'm not sure we need this extra check: `l.lines = l.lines[:0]` should be suffice – even if `l.lines == nil`. It is better to allocate the slice lazily, on append, not on `reset` (at least one allocation is avoided)

Reply:
I dont think I understand the comment. 
It looks like here I know the number of locations and the number of lines is at least the same, here I pre-allocate twice as much of locations in hope I never allocate again at all. I agree 2x is maybe too agressive for non arena allocations but for arena allocations it should not matter. At the same time if I allocate lazilly at append I will allocate multiple times: 2, 4, 8, 16, at least to next power of higher than nLocations.
So I dont understand in your comment which one allocation is avoided

Reply:
Never mind – I did get it wrong. Please consider using `grow` – in that case the intent would be more clear